File: problems/metaprogramming/metaprogramming.py
codes = {}

try:
    while True:
        line = input().split()

        if line[0] == "define":
            codes[line[2]] = int(line[1])
        else:
            a, b = line[1], line[3]
            op = line[2]

            if a not in codes or b not in codes:
                print("undefined")
                continue
            a = codes[a]
            b = codes[b]

            if op == ">":
                print("true" if a > b else "false")
            if op == "<":
                print("true" if a < b else "false")
            if op == "=":
                print("true" if a == b else "false")
except:
    pass

File: problems/metaprogramming/metaprogramming.cpp
#include 
#include 
#include 

using namespace std;

int main() {
    unordered_map codes;

    string nextCommand;

    while (cin >> nextCommand) {
        if (nextCommand == "define") {
            string key;
            int val;

            cin >> val >> key;
            codes[key] = val;
        } 
        else {
            string a, b, op;
            cin >> a >> op >> b;

            if (codes.find(a)== codes.end() || codes.find(b) == codes.end()) {
                cout << "undefined" << endl;
                continue;
            }

            int x = codes[a];
            int y = codes[b];

            if (op == ">") {
                cout << ((x > y) ? "true" : "false") << endl;
            } else if (op == "<") {
                cout << ((x < y) ? "true" : "false") << endl;
            }
            else {
                cout << ((x == y) ? "true" : "false") << endl;
            }
        }
    }

    
    return 0;
}